Migrating UI Automation to LabApiUtilities (MSAL)

fadidurah opened the pull request:
**What**
https://identitydivision.visualstudio.com/Engineering/_boards/board/t/Auth%20Client%20-%20Android/Backlog%20items/?workitem=1377734

We are currently using testutils to communicate with the Lab Api for testing purposes. We are moving away from this because testutils has a dependency on common which can cause a circular dependency. Instead, start the using the improved LabApiUtilities library. For MSAL part of this change, update all msalautomationapp TestCases to use LabApiUtilities instead of testutils. This should be reviewed AFTER Common PR.

Common PR: https://github.com/AzureAD/microsoft-authentication-library-common-for-android/pull/1736
Broker PR: https://github.com/AzureAD/ad-accounts-for-android/pull/1886

**Why**
Circular dependency in testutils will likely cause issue down the line, so earlier we can fix it the better. LabApiUtilities also contains some structural improvements in how we can communicate with Lab Api.

**How**
Update build.gradle to use LabApiUtlities instead of testutils, remove all references of testutils in TestCases, update any abstract classes to use LabApiUtilities implementations of communicating with Lab Api.

**Test**
Once the app is migrated to using LabApiUtilities, the success rate of the automated test cases should not be any lower than it was before the changes were introduced, and no test cases should be broken by these changes. Work to fix already broken test cases will be completed in the next sprint.

fadidurah committed:
infra work for LabApiUtilities, update common

fadidurah committed:
infra work for LabApiUtilities, update common

fadidurah committed:
Merge branch 'dev' of https://github.com/AzureAD/microsoft-authentication-library-for-android into fadi/LabUpi

fadidurah committed:
Fixed reference to Client Secret, update common, Fixed more cases

fadidurah committed:
Cleanup build gradle

fadidurah committed:
update common

fadidurah committed:
update changelog, update common

fadidurah committed:
fixed common branch

fadidurah committed:
update common

fadidurah committed:
Merge branch 'dev' of https://github.com/AzureAD/microsoft-authentication-library-for-android into fadi/LabUpi

shahzaibj commented on msalautomationapp/build.gradle:
```diff
@@ -182,7 +182,7 @@ dependencies {
 
-    implementation(project(":testutils")) {
+    implementation(project(":LabApiUtilities")) {
         exclude module: 'common'
```
I think we can remove this exclude now since LabApiUtilitities doesn't actually take a dependency on common

shahzaibj commented:
The suggestions I provided in the other 2 PRs (common and broker) will apply here as as well, so please consider those for this PR as well.

somalaya commented on msalautomationapp/src/androidTest/java/com/microsoft/identity/client/msal/automationapp/testpass/broker/TestCase833513.java:
```diff
@@ -77,12 +76,14 @@ public void test_833513() throws MsalException, InterruptedException {
 
-        final String username = LabUserHelper.loadUserForTest(query);
-        String password = LabConfig.getCurrentLabConfig().getLabUserPassword();
+        final LabAccount user2 = mLabClient.getLabAccount(query);
```
nit : var name can be user

somalaya commented on msalautomationapp/src/androidTest/java/com/microsoft/identity/client/msal/automationapp/testpass/broker/TestCase833516.java:
```diff
@@ -72,11 +74,13 @@ public void test_833516() throws MsalException, InterruptedException {
 
-        final String username = LabUserHelper.loadUserForTest(query);
-        String password = LabConfig.getCurrentLabConfig().getLabUserPassword();
+        final LabAccount user2 = mLabClient.getLabAccount(query);
```
var can be user?

somalaya commented on msalautomationapp/src/androidTest/java/com/microsoft/identity/client/msal/automationapp/testpass/broker/TestCase833516.java:
```diff
@@ -103,26 +107,28 @@ public void test_833516() throws MsalException, InterruptedException {
 
-        final String anotherUserFromSameTenant = LabUserHelper.loadUserForTest(query2);
-        password = LabConfig.getCurrentLabConfig().getLabUserPassword();
+        final LabAccount user3 = mLabClient.getLabAccount(query2);
```
can be user2?

somalaya commented on msalautomationapp/src/androidTest/java/com/microsoft/identity/client/msal/automationapp/testpass/broker/crosscloud/TestCase1592468.java:
```diff
@@ -70,7 +73,7 @@ public TestCase1592468(final String name, final String guestHomeAzureEnvironment
     public static Collection guestHomeAzureEnvironment() {
         return Arrays.asList(new Object[][]{
-                {"AZURE_US_GOV", LabConstants.GuestHomeAzureEnvironment.AZURE_US_GOV, /*homeCloud*/"https://login.microsoftonline.us", /*crossCloud*/"https://login.microsoftonline.com"},
+                {"AZURE_US_GOV", GuestHomeAzureEnvironment.AZURE_US_GOVERNMENT, /*homeCloud*/"https://login.microsoftonline.us", /*crossCloud*/"https://login.microsoftonline.com"},
```
should this be GuestHomeAzureEnvironment.AZURE_US_GOVERNMENT.name() instead of just GuestHomeAzureEnvironment.AZURE_US_GOVERNMENT

fadidurah replied:
I think it should be toString(), actually. But yes. Stuff like this should show up once I re run automation.

fadidurah committed:
Changes for comments, update common

fadidurah committed:
update common, fix AbstractB2CTest reference

fadidurah committed:
address changes in common, update common

fadidurah committed:
merge shahzaib's changes

fadidurah committed:
update common

fadidurah committed:
update common

fadidurah committed:
androidx version update

shahzaibj commented on msalautomationapp/src/androidTest/java/com/microsoft/identity/client/msal/automationapp/testpass/broker/crosscloud/TestCase1592468.java:
```diff
@@ -137,13 +139,13 @@ public void test_acquireTokenFromCrossCloudForJoinedAccount() throws Throwable {
-        query.guestHomedIn = LabConstants.GuestHomedIn.HOST_AZURE_AD;
-        query.azureEnvironment = LabConstants.AzureEnvironment.AZURE_CLOUD;
-        return query;
+    public LabQuery getLabQuery() {
+        return LabQuery.builder()
+                .userType(UserType.GUEST)
+                .guestHomeAzureEnvironment(GuestHomeAzureEnvironment.valueOf(mGuestHomeAzureEnvironment))
+                .guestHomedIn(GuestHomedIn.HOST_AZURE_AD)
+                .azureEnvironment(AzureEnvironment.AZURE_CLOUD)
+                .build();
```
the formatting looks a bit off here

fadidurah replied:
This is how android studio auto indents it. how would you format instead?

shahzaibj replied:
Looks like it is okay, it was just rendering in a weird way in GitHub: 

![image](https://user-images.githubusercontent.com/37125644/170648220-ac68beb6-6825-47dd-be6f-898212ac902f.png)

shahzaibj commented on msalautomationapp/src/androidTest/java/com/microsoft/identity/client/msal/automationapp/testpass/broker/crosscloud/TestCase1592465.java:
```diff
@@ -123,13 +125,13 @@ public void test_acquire_token_from_cross_cloud_after_acquiring_token_from_home_
-        query.guestHomedIn = LabConstants.GuestHomedIn.HOST_AZURE_AD;
-        query.azureEnvironment = LabConstants.AzureEnvironment.AZURE_CLOUD;
-        return query;
+    public LabQuery getLabQuery() {
+        return LabQuery.builder()
+                .userType(UserType.GUEST)
+                .guestHomeAzureEnvironment(GuestHomeAzureEnvironment.valueOf(mGuestHomeAzureEnvironment))
+                .guestHomedIn(GuestHomedIn.HOST_AZURE_AD)
+                .azureEnvironment(AzureEnvironment.AZURE_CLOUD)
+                .build();
```
formatting looks a bit off

shahzaibj commented on msalautomationapp/src/androidTest/java/com/microsoft/identity/client/msal/automationapp/testpass/broker/crosscloud/TestCase1420494.java:
```diff
@@ -116,13 +118,13 @@ public void test_1420494() throws Throwable {
-        query.guestHomedIn = LabConstants.GuestHomedIn.HOST_AZURE_AD;
-        query.azureEnvironment = LabConstants.AzureEnvironment.AZURE_CLOUD;
-        return query;
+    public LabQuery getLabQuery() {
+        return LabQuery.builder()
+                .userType(UserType.GUEST)
+                .guestHomeAzureEnvironment(GuestHomeAzureEnvironment.valueOf(mGuestHomeAzureEnvironment))
+                .guestHomedIn(GuestHomedIn.HOST_AZURE_AD)
+                .azureEnvironment(AzureEnvironment.AZURE_CLOUD)
+                .build();
```
formatting looks a bit off

shahzaibj commented on msalautomationapp/src/androidTest/java/com/microsoft/identity/client/msal/automationapp/testpass/broker/crosscloud/TestCase1592510.java:
```diff
@@ -123,13 +125,13 @@ public void test_acquire_token_from_home_cloud_after_acquiring_token_from_cross_
-        query.guestHomedIn = LabConstants.GuestHomedIn.HOST_AZURE_AD;
-        query.azureEnvironment = LabConstants.AzureEnvironment.AZURE_CLOUD;
-        return query;
+    public LabQuery getLabQuery() {
+        return LabQuery.builder()
+                .userType(UserType.GUEST)
+                .guestHomeAzureEnvironment(GuestHomeAzureEnvironment.valueOf(mGuestHomeAzureEnvironment))
+                .guestHomedIn(GuestHomedIn.HOST_AZURE_AD)
+                .azureEnvironment(AzureEnvironment.AZURE_CLOUD)
+                .build();
```
formatting looks a bit off

shahzaibj commented on msalautomationapp/src/androidTest/java/com/microsoft/identity/client/msal/automationapp/testpass/crosscloud/TestCase1616315.java:
```diff
@@ -136,13 +138,13 @@ public void test_acquire_token_from_cross_cloud_after_acquiring_token_from_home_
-        query.guestHomedIn = LabConstants.GuestHomedIn.HOST_AZURE_AD;
-        query.azureEnvironment = LabConstants.AzureEnvironment.AZURE_CLOUD;
-        return query;
+    public LabQuery getLabQuery() {
+        return LabQuery.builder()
+                .userType(UserType.GUEST)
+                .guestHomeAzureEnvironment(GuestHomeAzureEnvironment.valueOf(mGuestHomeAzureEnvironment))
+                .guestHomedIn(GuestHomedIn.HOST_AZURE_AD)
+                .azureEnvironment(AzureEnvironment.AZURE_CLOUD)
+                .build();
```
formatting looks a bit off

shahzaibj commented on msalautomationapp/src/androidTest/java/com/microsoft/identity/client/msal/automationapp/testpass/labclient/LabClientResetPasswordTest.java:
```diff
@@ -0,0 +1,125 @@
+import java.util.Arrays;
+import java.util.concurrent.TimeUnit;
+
+public class LabClientResetPasswordTest extends AbstractMsalBrokerTest {
```
Is this a test case in ADO for MSAL? or just something you're doing to test Lab Api?

fadidurah replied:
Just testing Lab Api, I placed i in automationapp so I could use MsalSdk, it could probably go somewhere else.

shahzaibj replied:
You may choose to just create a test case in ADO for this and then this gets a test id and becomes a lab client password reset test + also an MSAL test at the same time

shahzaibj commented on msalautomationapp/src/androidTest/java/com/microsoft/identity/client/msal/automationapp/testpass/local/TestCase497038.java:
```diff
@@ -83,7 +82,7 @@ public void test_497038() throws Throwable {
 
         // sleep as it can take a bit for UPN to appear in Azure Sample app
-        Thread.sleep(TimeUnit.SECONDS.toMillis(5));
+        Thread.sleep(TimeUnit.SECONDS.toMillis(10));
```
seems like it takes much longer

fadidurah replied:
For the upn to appear? I can increase the sleep time

shahzaibj replied:
I meant you already increased it, so I assume it was taking longer than 5 seconds for you and hence you increased it to 10 seconds?

fadidurah replied:
I think this is left over code I had. from my test passes it looks like it was still failing with 10 seconds. I'll probably look into this more next sprint.

fadidurah committed:
Update TestCase497038.java

revert second increase as test still failed, will take a look once I commit to fixing test cases in june.
